js/ui/windowManager.js

```diff
@@ -2140,66 +2179,51 @@ const CodingManager = new Lang.Class({
         let focusedWindow = global.display.get_focus_window();
         if (!focusedWindow)
             return;
-        if (!session.windowApp.get_meta_window())
-            return;
 
-        if (session.buttonApp)
-            session.buttonApp.hide();
-        if (session.buttonBuilder)
-            session.buttonBuilder.hide();
+        let previousFocused = this._previousFocusedWindow;
+        this._previousFocusedWindow = focusedWindow;
 
-        if (focusedWindow.get_stable_sequence() === session.windowApp.get_meta_window().get_stable_sequence()){
-            if (session.state === SessionState.APP) {
-                session.buttonApp.show();
-                return;
+        session.buttonApp.hide();
```

Comment:
Perhaps but this in an `else` statement below. Right now we hide the button only to show it immediately afterwards if there is no corresponding builder window.